src/block/BaseSign.php

```diff
@@ -37,14 +40,15 @@
 use function strlen;
 
 abstract class BaseSign extends Transparent{
-	//TODO: conditionally useless properties, find a way to fix
-
 	protected SignText $text;
 	protected ?int $editorEntityRuntimeId = null;
+	protected DyeColor $textColor;
```

Comment:
This needs to be a `Color`, not a `DyeColor`, to allow plugins access to the full range of options available.

The default should be `0, 0, 0`.